### PR TITLE
chore: add support for a local devnet

### PR DESCRIPTION
**Pull request by rach-id**
## Overview

When trying to play with some WAL related work, I had to build this simple network to be able to run a multi validator network locally and experiment with it.

This network provides a way to run fast experiments locally on celestia-app and fills the gap with needing to rely on Knuu (and all the docker images that need to be created) or testground (which require a lot of resources and not so straightforward to run). With this, you only need docker-compose and you have a four validators network.

Also, it provides out of the box telemetry, prometheus/grafana, already setup for all the validators. This also allows seeing the metrics without having to resort to any remote servers etc.

So, I decided to include this in app, if it's not that needed, we can keep it in a separate branch, or I can keep using it only locally :D

### CODE REVIEW

**Review by rootulp — DISMISSED** [review]
This is awesome, great work! I may try to use this instead of knuu for the straggling node test: https://github.com/celestiaorg/celestia-app/issues/3297. I think I'll need to modify the genesis so that the chain starts on app version 1... will try and report back

**Review by evan-forbes — APPROVED** [review]
**definitely not blocking**, but we might want to track some potential changes upon merging.

use tooling similar to what we use in knuu, testground, or the core e2e test where we generate the files instead of keeping them in a directory. This way, its easier to confidently make changes across the entire network, make changes such the number of nodes, and there's less things in the repo.

this is definitely an overdue very handy tool! thanks for taking the time adapt the work done in blobstream for here!

File: local_devnet/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+docker-compose down
+```
+
+## Monitoring
+
+Monitoring is preconfigured in the `celestia-app/config.toml`. This means that you will have access to the metrics the moment you spin up the devnet along with telemetry.
+
+### Accessing grafana
+
+Grafana is exposed in `localhost:3000`. The default credentials are `admin:admin`. Then, you will find predefined data sources to get the metrics from.
```
[the remark this answers is not in the document]
Comment (rach-id) [reply local_devnet/README.md:108]: Good idea, I'll preload some

**Review by rootulp — APPROVED** [review]
This saved me so much time while working on straggling node test, thanks!

**Comment by rach-id** [comment]
> This way, its easier to confidently make changes across the entire network, make changes such the number of nodes, and there's less things in the repo.

What I am thinking of is:

- This local devnet to be minimal, all the files out (so that it's easy to change and restart), and only support tests that are `[1, 4]` validators. I don't think it makes sense to run a network of 10 validators locally for anyone, or at least not that often.
- My next tool, which I am still thinking of a good way to have it will create the files on the fly, and integrate other stuff so that we're able to spin up bigger devnets. I might even add it to Knuu to re-use what we already have.

I can open an issue for the latter to track it.

**Review by evan-forbes — APPROVED** [review]

File: local_devnet/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rach-id) [reply local_devnet/README.md:108]: Fixed port in 656016395fb6ef42aab494309f90b80260616bb4

**Comment by evan-forbes** [comment]
> What I am thinking of is:
This local devnet to be minimal, all the files out (so that it's easy to change and restart), and only support tests that are [1, 4] validators. I don't think it makes sense to run a network of 10 validators locally for anyone, or at least not that often.
My next tool, which I am still thinking of a good way to have it will create the files on the fly, and integrate other stuff so that we're able to spin up bigger devnets. I might even add it to Knuu to re-use what we already have.
I can open an issue for the latter to track it.

this makes sense, however I do think we should only have a single too ideally. This only the fly generation also overlaps heavily with what we do in knuu, so perhaps we should only knuu and minikube for this type of test. The only thing we have to do here is open and issue to save this context. we certainly don't have to make a decision now, just document this is all :slightly_smiling_face:

**Comment by rach-id** [comment]
Sounds good 👍 Thanks

**Review by rootulp — APPROVED** [review]

**Pull request merged**